File: source/Knyaz.Optimus.JsTests/Tests/DomParserTests.js
Test("DomParserTests", {
	"DomParserType":{
		run: function(){
			Assert.AreEqual("function", typeof DOMParser);
		}
	},
	"DomParserConstructor": {
		run: function () {
			var parser = new DOMParser();
			Assert.IsNotNull(parser.parseFromString);
		}
	},
	"DomParserWindowConstructor": {
		run: function () {
			var parser = new window.DOMParser();
			Assert.IsNotNull(parser.parseFromString);
		}
	},
	"ParseHtml":{
		run: function () {
			var parser = new DOMParser();
			var doc = parser.parseFromString("<html><title>asd</title></html>", "text/html");
			Assert.IsNotNull(doc);
			Assert.AreEqual("asd", doc.title);
		}
	}
});

File: source/Knyaz.Optimus.UnitTests/Dom/DomParserTests.cs
using Knyaz.Optimus.Dom;
using NUnit.Framework;

namespace Knyaz.Optimus.Tests.Dom
{
	[TestFixture]
	public class DomParserTests
	{
		[Test]
		public static void ParseHtml()
		{
			var parser = new DomParser();
			var doc = (HtmlDocument)parser.ParseFromString("<html><title>qwe</title></html>", "text/html");
			Assert.AreEqual("qwe", doc.Title);
		}
	}
}